packages/core/core-flows/src/product/workflows/update-products.ts

```diff
@@ -20,7 +33,60 @@ export const updateProductsWorkflow = createWorkflow(
     input: WorkflowData<WorkflowInput>
   ): WorkflowData<ProductTypes.ProductDTO[]> => {
     // TODO: Delete price sets for removed variants
-    // TODO Update sales channel links
-    return updateProductsStep(input)
+
+    const toUpdateInput = transform({ input }, (data) => {
+      if ((data.input as UpdateProductsStepInputProducts).products) {
+        return (data.input as UpdateProductsStepInputProducts).products.map(
+          (p) => ({
+            ...p,
+            sales_channels: undefined,
+          })
+        )
+      } else {
+        return {
+          selector: (data.input as UpdateProductsStepInputSelector).selector,
+          update: {
+            ...(data.input as UpdateProductsStepInputSelector).update,
+            sales_channels: undefined,
+          },
+        }
+      }
+    })
+
+    const updatedProducts = updateProductsStep(toUpdateInput as any) // TODO: type
+
+    const salesChannelLinks = transform({ input, updatedProducts }, (data) => {
+      if ((data.input as UpdateProductsStepInputSelector).selector) {
+        if (
+          !(data.input as UpdateProductsStepInputSelector).update.sales_channels
+        ) {
+          return []
+        }
+        return data.updatedProducts.map((p) => {
+          return {
+            product_id: p.id,
+            sales_channel_ids: (
+              (data.input as UpdateProductsStepInputSelector).update
```

Comment:
NIT: you can do `data.input.update.sales_channels?.map(c => c.id) ?? []` instead, it's a bit easier to read.